src/main/java/de/tum/in/www1/artemis/service/connectors/VersionControlService.java

```diff
@@ -110,5 +112,16 @@
      * @param projectName to check if a project with the same name already exists
      * @return true if the project exists, false otherwise
      */
-    public String checkIfProjectExists(String projectKey, String projectName);
+    String checkIfProjectExists(String projectKey, String projectName);
+
+    /**
+     * Copies a repository from one project to another one. The project can be the same.
+     *
+     * @param baseRepositoryUrl The URL of the template repository
+     * @param targetProjectKey The key of the target project
+     * @param targetRepositorySlug The slug of the new repository
+     * @param username Can be null. The name of the user if the newly created repository should belong to a specific user
+     * @return A map containing various information about the copied repository, s.a. the slug or the cloneUrl of the new repo.
+     */
+    Map<String, String> copyRepository(URL baseRepositoryUrl, String targetProjectKey, String targetRepositorySlug, @Nullable String username);
```

Comment:
please change the signature of this method to something like

VCSRepository copyRepository(String sourceProjectName, String sourceRepositoryName, String targetProjectName, String targetRepositoryName)

Reply:
I refactored the signature